File: pset6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Get text
    text = get_string("Text: ")
    # Get number of letters, words, and sentences
    numLetters = count_letters(text)
    numWords = count_words(text)
    numSentences = count_sentences(text)
    # Computer L (number of letters per 100 words) and S (number of sentences per 100 words)
    L = (numLetters / numWords) * 100
    S = (numSentences / numWords) * 100
    # Compute the index using L and S
    index = 0.0588 * L - 0.296 * S - 15.8
    # Round the index to get the grade
    grade = round(index)
    # Print the grade
    if grade >= 1 and grade < 16:
        print("Grade " + str(grade))
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")


def count_letters(text):
    # Intantiate letter count
    letterCount = 0
    # Iterate over each character
    for c in text:
        # If a character is a letter, add to the letter count
        if c.isalpha():
            letterCount += 1
    return letterCount


def count_words(text):
    # Intantiate word count
    wordCount = 0
    # Iterate over each character
    for c in text:
        # If there is a space, we know the characters before it were a word
        if c == ' ':
            wordCount += 1
    # Add one in the end to account for the last word - no space after
    return wordCount + 1


def count_sentences(text):
    # Intantiate sentence count
    sentenceCount = 0
    # Iterate over each character
    for c in text:
        # If there is a . ! ? then it is ending a sentence, so we add to sentecne counter
        if c == '.' or c == '!' or c == '?':
            sentenceCount += 1
    return sentenceCount


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void) {
    //Get the text
    string input = get_string("Text: ");
    //Get the number of lettters, words, and sentences
    int numLetters = count_letters(input);
    int numWords = count_words(input);
    int numSentences = count_sentences(input);

    //Compute L (number of letters per 100 words) and S (number of sentences per 100 words)
    double L = ((double) numLetters / numWords) * 100;
    double S = ((double) numSentences / numWords) * 100;
    //Compute the index using L and S
    double index = 0.0588 * L - 0.296 * S - 15.8;
    //Round the index to get the grade
    int grade = (int) round(index);

    //Print the grade
    if (grade >= 1 && grade < 16) {
        printf("Grade %i\n", grade);
    }
    else if (grade >= 16) {
        printf("Grade 16+\n");
    }
    else {
        printf("Before Grade 1\n");
    }
}

int count_letters(string text) {
    //Find length and intantiate letter count
    int length = strlen(text);
    int letterCount = 0;
    //Iterate over each character
    for (int i = 0; i < length; i++) {
        //If a character is a letter, add to the letter count
        if (islower(text[i]) || isupper(text[i])) {
            letterCount++;
        }
    }
    return letterCount;
}

int count_words(string text) {
    //Find length and intantiate word count
    int length = strlen(text);
    int wordCount = 0;
    //Iterate over each character
    for (int i = 0; i < length; i++) {
        //If there is a space, we know the characters before it were a word
        if (text[i] == ' ') {
            wordCount++;
        }
    }
    //Add one in the end to account for the last word - no space after
    return wordCount + 1;
}

int count_sentences(string text) {
    //Find length and intantiate sentence count
    int length = strlen(text);
    int sentenceCount = 0;
    //Iterate over each character
    for (int i = 0; i < length; i++) {
        //If there is a . ! ? then it is ending a sentence, so we add to sentecne counter
        if (text[i] == '.' || text[i] == '!' || text[i] == '?') {
            sentenceCount++;
        }
    }
    return sentenceCount;
}
